src/client.js

```diff
@@ -248,14 +382,20 @@ class TouchPortalClient extends EventEmitter {
     this.send(pairMsg);
   }
 
-  checkForUpdate() {
+  /**
+   * checkForUpdate() A method that http request
+   * to a Json that contains `version` key to compare.
+   * 
+   * @param {String} updateUrl - webpage url contains json
+   * 
+   * @return {void}
+   */
+  checkForUpdate(updateUrl) {
     const parent = this;
-    http.get(this.updateUrl, (res) => {
+    http.get(updateUrl, (res) => {
       const { statusCode } = res;
 
-      // Any 2xx status code signals a successful response but
-      // here we're only checking for 200.
-      if (statusCode !== 200) {
+      if (statusCode >= 200 && statusCode <= 299) {
```

Comment:
this needs to change to statusCode < 200 || statusCode > 299 if we are going to stick with using any 200 level response as a success.  Right now this will error out when successful instead of skipping this block.
```suggestion
      if (statusCode < 200 || statusCode > 299) {
```

Reply:
ah sorry. mb didn't realize it lol